### PR TITLE
feature: define api types for seed pattern

### PR DESCRIPTION
**Pull request by wangforthinker**
Signed-off-by: allen.wq <allen.wq@alipay.com>

### Ⅰ. Describe what this PR did
define some api types for seed mode.

### Ⅱ. Does this pull request fix one issue?

NONE.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
none.

### CODE REVIEW

File: apis/swagger.yml [hunk 1]
```diff
@@ -1362,6 +1394,93 @@ definitions:
...
+          $ref: "#/definitions/PieceInfo"
+
+  Node:
+    type: "object"
+    description: ""
+    properties:
+      basic:
+        $ref: "#/definitions/PeerInfo"
+        description: "Basic node info"
+      extra:
```
Comment (lowzj) [inline apis/swagger.yml:1440]: I think the `extra` property is a map struct not a pre-defined `object`, thus we can add some customized properties without changing the `api` in the future.

File: apis/swagger.yml [hunk 2]
```diff
@@ -1362,6 +1394,93 @@ definitions:
         format: "int64"
         description: "The length of the file dfget requests to download in bytes."
 
+  NetworkInfoFetchRequest:
+    type: "object"
+    description: ""
```
Comment (allencloud) [inline apis/swagger.yml:1399]: Please add description here.

File: apis/swagger.yml [hunk 3]
```diff
@@ -1362,6 +1394,93 @@ definitions:
...
+      urls:
+        type: "array"
+        description: |
+          The urls is to filter the peer node, the url should be match with taskURL in TaskInfo.
+        items:
+          type: "string"
+
+  Extra:
+    type: "object"
+    description: ""
```
Comment (allencloud) [inline apis/swagger.yml:1410]: Please add description here.

File: apis/swagger.yml [hunk 4]
```diff
@@ -1362,6 +1394,93 @@ definitions:
...
+      room:
+        type: "string"
+      rack:
+        type: "string"
+      idc:
+        type: "string"
+
+  TaskFetchInfo:
+    type: "object"
+    description: ""
```
Comment (allencloud) [inline apis/swagger.yml:1423]: Please add description here.

File: apis/swagger.yml [hunk 5]
```diff
@@ -1362,6 +1394,93 @@ definitions:
...
+        $ref: "#/definitions/TaskInfo"
+      pieces:
+        type: "array"
+        description: "The pieces which should belong to the peer node"
+        items:
+          $ref: "#/definitions/PieceInfo"
+
+  Node:
+    type: "object"
+    description: ""
```
Comment (allencloud) [inline apis/swagger.yml:1435]: Please add description here.

File: apis/swagger.yml [hunk 6]
```diff
@@ -945,6 +961,18 @@ definitions:
...
+        description: |
+          This attribute represents the digest of resource, dfdaemon or dfget catches this parameter
+          from the headers of request URL. The digest will be considered as the taskID if not null.
+      fileLength:
+        type: "integer"
+        description: |
+          This attribute represents the length of resource, dfdaemon or dfget catches and calculates
+          this parameter from the headers of request URL. If fileLength is vaild, the supernode need
+          not get the length of resource by accessing the rawURL.
+          format: "int64"
```
Comment (allencloud) [inline apis/swagger.yml:975]: Maybe the indent here is incorrect.

File: apis/swagger.yml [hunk 7]
```diff
@@ -1362,6 +1394,93 @@ definitions:
...
+          $ref: "#/definitions/Node"
+
+  HeartBeatRequest:
+    type: "object"
+    description: ""
+    properties:
+      IP:
+        type: "string"
+        description: "IP address which peer client carries"
+        format: "string"
```
Comment (allencloud) [inline apis/swagger.yml:1466]: Please use `ipv4` or `ipv6` for the format here.

File: apis/swagger.yml [hunk 8]
```diff
@@ -1362,6 +1394,93 @@ definitions:
...
+  NetworkInfoFetchResponse:
+    type: "object"
+    description: ""
+    properties:
+      nodes:
+        type: "array"
+        items:
+          $ref: "#/definitions/Node"
+
+  HeartBeatRequest:
```
Comment (allencloud) [inline apis/swagger.yml:1459]: This struct is newly declared here, while where to use it in the API endpoint?
I think where to use it is missing in the API defining side.

**Comment by lowzj** [comment]
```
apis/types/task_register_request.go:110:28: "specfic" is a misspelling of "specific"
apis/swagger.yml:799:34: "specfic" is a misspelling of "specific"
apis/types/task_register_request.go:110:28: "specfic" is a misspelling of "specific"
apis/types/task_register_request.go:110:28: "specfic" is a misspelling of "specific"
./apis/types/task_register_request.go:110:28: "specfic" is a misspelling of "specific"
apis/swagger.yml:799:34: "specfic" is a misspelling of "specific"
./apis/swagger.yml:799:34: "specfic" is a misspelling of "specific"
```
please resolve these spell errors

File: apis/swagger.yml (hunk 1 above)
Comment (wangforthinker) [reply apis/swagger.yml:1468]: done.

File: apis/swagger.yml (hunk 2 above)
Comment (wangforthinker) [reply apis/swagger.yml]: done.

File: apis/swagger.yml (hunk 3 above)
Comment (wangforthinker) [reply apis/swagger.yml]: done.

File: apis/swagger.yml (hunk 4 above)
Comment (wangforthinker) [reply apis/swagger.yml]: done.

File: apis/swagger.yml (hunk 5 above)
Comment (wangforthinker) [reply apis/swagger.yml]: done.

File: apis/swagger.yml (hunk 6 above)
Comment (wangforthinker) [reply apis/swagger.yml]: done.

File: apis/swagger.yml (hunk 7 above)
Comment (wangforthinker) [reply apis/swagger.yml]: done.

File: apis/swagger.yml (hunk 8 above)
Comment (wangforthinker) [reply apis/swagger.yml:1489]: done.

File: apis/types/extra.go [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+import (
+	strfmt "github.com/go-openapi/strfmt"
+
+	"github.com/go-openapi/swag"
+)
+
+// Extra extra
+// swagger:model Extra
+type Extra struct {
```
Comment (lowzj) [inline apis/types/extra.go:16]: This file is not generated by  go-swagger with the current api definition.

Comment (wangforthinker) [reply apis/types/extra.go]: done.

File: apis/swagger.yml [hunk 10]
```diff
@@ -648,6 +648,47 @@ paths:
         500:
           $ref: "#/responses/500ErrorResponse"
 
+  /peer/heartbeat:
+    get:
```
Comment (allencloud) [inline apis/swagger.yml:652]: I do not think this is a "GET" request. Maybe `POST` is better.  @wangforthinker

**Pull request merged**